src/core/tcp_in.c

```diff
@@ -608,7 +608,9 @@ tcp_input(struct pbuf *p, struct ip_globals *ip_data)
     pbuf_free(p);
   }
 
+  SYS_ARCH_LOCK(&tcp_mutex);
   LWIP_ASSERT("tcp_input: tcp_pcbs_sane()", tcp_pcbs_sane());
+  SYS_ARCH_UNLOCK(&tcp_mutex);
```

Comment:
The tcp_pcbs_sane() function exists only `#if TCP_DEBUG || TCP_INPUT_DEBUG || TCP_OUTPUT_DEBUG`, so perhaps we could put the above 3 lines under `#if TCP_INPUT_DEBUG`? Otherwise in non-debug builds we are locking and unlocking tcp_mutex unnecessarily.

Reply:
Good point, fixed.